src/ast/semantic_analyser.cpp

```diff
@@ -157,9 +157,11 @@ void SemanticAnalyser::visit(Builtin &builtin)
     for (auto &attach_point : *probe_->attach_points)
     {
       ProbeType type = probetype(attach_point->provider);
-      if (type != ProbeType::tracepoint)
-        err_ << "The args builtin can only be used with tracepoint probes"
+      if (type != ProbeType::tracepoint) {
+        err_ << "The args builtin can only be used with tracepoint probes "
              << "(" << attach_point->provider << " used here)" << std::endl;
+        return;
```

Comment:
```suggestion
        continue;
```

So if args is used in more than one probe, we warn the user of all invalid uses (as in https://github.com/iovisor/bpftrace/pull/724)